types/got/index.d.ts

```diff
@@ -202,7 +202,7 @@ declare namespace got {
         baseUrl?: string;
         cookieJar?: CookieJar;
         encoding?: E;
-        query?: string | object;
+        query?: string | Record<string, string | number> | URLSearchParams;
```

Comment:
As I mentioned before, `Record<string, string | number>` is breaking change. Type in documentation is just a hint, user can still pass any type and it will be converted to string. I think more appropriate type here is `Record<string, any>`. Also, I don't understand  why `number` should be excluded?

For example, `querystring.stringify()` from `Node.js` declaration takes any type: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/46014b89d69bf012809c9190374d883204516193/types/node/querystring.d.ts#L17

I want to hear opinions of other maintainers.